hil/config.py

```diff
@@ -8,13 +8,85 @@
 import ConfigParser
 import logging.handlers
 import importlib
+from schema import Schema, Optional
 import os
 import sys
+from urlparse import urlparse
 
 cfg = ConfigParser.RawConfigParser()
 cfg.optionxform = str
 
 
+def string_is_bool(option):
+    """Check if a string matches ConfigParser's definition of a bool"""
+    return option.lower() in ['true', 'yes', 'on', '1',
+                              'false', 'no', 'off', '0']
+
+def string_is_web_url(option):
+    """Check if a string is a valid web URL"""
+    url = urlparse(option)
+    if url.scheme == '' or url.netloc == '':
+        return False
+    return True
+
+def string_is_db_uri(option):
+    """Check if a string is a valid DB URI"""
+    url = urlparse(option)
+    if not ('postgres' in url.scheme or 'sqlite' in url.scheme):
+        return False
+    if url.path == '':
+        return False
+    return True
+
+def is_log_level(option):
+    return option.lower() in ['debug', 'info', 'warn', 'warning', 'error',
+                      'critical', 'fatal']
+
+def string_has_vlans(option):
+    """Check if a string is a valid list of VLANs"""
+    for r in option.split(","):
+        r = r.strip().split("-")
+        if not all(s.isdigit() for s in r):
```

Comment:
1. you can actually just call `isdigit` once:

```python
>>> '423'.isdigit()
True
```

2. Why not actually check the range?

```python
s.isdigit() and 0 < int(s) <= 4096
```

Or, for that matter, just lift the relevant bit of the schema from the dummy_vlan field.

Reply:
I'll check the range here, good for a first safety check

Reply:
Ah, you're right, I was thinking you were checking each character.